lib/ansible/plugins/filter/urlsplit.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+from ansible.module_utils.six.moves.urllib.parse import urlsplit
+from ansible.errors import AnsibleFilterError
+
+
+def split_url(value, query='', alias='urlsplit'):
+    ''' Check if string is an URI and filter it '''
+
+    results = urlsplit(value)
+
+    split_url = dict(
+        fragment=results.fragment,
+        hostname=results.hostname,
+        password=results.password,
+        path=results.path,
+        port=results.port,
+        query=results.query,
+        scheme=results.scheme,
+        username=results.username
+    )
```

Comment:
you could just use this function instead https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/helpers.py#L37